src/schema/v2/lot.ts

```diff
@@ -0,0 +1,179 @@
+import { GraphQLError, GraphQLObjectType } from "graphql"
+import { SaleArtworkType } from "schema/v2/sale_artwork"
+import { Response as FetchResponse } from "node-fetch"
+import { ResolverContext } from "types/graphql"
+import {
+  NodeInterface,
+  SlugAndInternalIDFields,
+} from "schema/v2/object_identification"
+import { pageable } from "relay-cursor-paging"
+import { connectionWithCursorInfo } from "./fields/pagination"
+import gql from "lib/gql"
+import { convertConnectionArgsToGravityArgs } from "lib/helpers"
+import { connectionFromArray } from "graphql-relay"
+
+export const LotType = new GraphQLObjectType<any, ResolverContext>({
+  name: "Lot",
+  description:
+    "A lot in an auction containing merged Sale artwork and Lot state data",
+  interfaces: () => {
+    return [NodeInterface]
+  },
+  fields: () => {
+    return {
+      // Must place an `id` at the root of the object (next to saleArtwork)
+      // when resolving a lot for this to work (see resolver)
+      ...SlugAndInternalIDFields,
+      saleArtwork: {
+        type: SaleArtworkType,
+        description: "The saleArtwork object.",
+        resolve: ({ saleArtwork }) => {
+          return saleArtwork
+        },
+      },
+    }
+  },
+})
+
+/**
+ * Fetch relevant sale artworks and use them to fetch lot states
+ * from causality. Then pass the lot states through to the stitching layer
+ * via the graphql request context.
+ */
+const resolveWatchedLotConnection = async (_parent, args, context) => {
+  const { saleArtworksAllLoader, causalityLoader } = context
+  // fetch sale artworks from gravity
+  const { first = 25, ...rest } = args
+
+  const connectionOptions = {
+    include_watched_artworks: true,
+    total_count: true,
+    first,
+    ...rest,
+  }
+  const params = convertConnectionArgsToGravityArgs(connectionOptions)
+  delete params.page
+
+  const { body, headers } = await saleArtworksAllLoader(params)
+  const watchedSaleArtworks: any[] = body
+  const totalCount = parseInt(headers["x-total-count"] || "0", 10)
+  const saleArtworkIds = watchedSaleArtworks.map((sa) => sa._id)
```

Comment:
(non-blocking tip) - This line is a pretty contrived example since its a one-liner, but have noticed this elsewhere. It's much better to be verbose and use real words with variables, versus abbreviate, as abbreviations make it hard to reason about code through time, especially when another dev picks it up. 

```ts
const saleArtworkIds = watchedSaleArtworks.map((saleArtwork) => saleArtwork._id)
```

Reply:
yeah, sorry. this is a common variable naming pattern in scala but maybe scala is not the best language to take naming patterns from.